src/ParallelAlgorithms/Amr/Criteria/Criterion.hpp

```diff
@@ -20,9 +20,23 @@ namespace amr {
 /// \brief Base class for something that determines how an adaptive mesh should
 /// be changed
 ///
-/// \details When AMR criteria are evaluated for each element, they should
-/// return a std::array<amr::domain::Flag, Dim> containing the recommended
-/// refinement choice in each logical dimension of the Element.
+/// \details Each class derived from this class should (see the examples below):
+/// - Be option-creatable
+/// - Be serializable
+/// - Define a call operator that returns a std::array<amr::domain::Flag, Dim>
+///   containing the recommended refinement choice in each logical dimension of
+///   the Element.
+/// - Define the type aliases `argument_tags` and
+///   `compute_tags_for_observation_box` that are type lists of tags used in the
+///   call operator.
+/// The call operator should take as arguments the values corresponding to each
+/// tag in `argument_tags` (in order), followed by the Parallel::GlobalCache,
+/// and the ArrayIndex (which is typically an ElementId).  The tags listed in
```

Comment:
Can we simplify this and just make this an `ElementId`, removing the `ArrayIndex` template parameter? We assume it's an ElementId anyway down the line. Feel free to push as separate commit if you agree.

Reply:
okay done